### PR TITLE
nix 2.25

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: src/drv.cc [hunk 1]
```diff
@@ -63,6 +63,19 @@ auto queryCacheStatus(nix::Store &store,
...
+    nlohmann::json json;
+    for (const auto &pair : outputs) {
+        if (pair.second) {
+            json[pair.first] = pair.second.value();
+        } else {
+            json[pair.first] = nullptr;
+        }
+    }
+    return json;
+}
```
Comment (Ericson2314) [inline src/drv.cc:78]: I think with the right headers you shouldn't need this function?

Comment (Ericson2314) [reply src/drv.cc:78]: https://github.com/NixOS/nix/blob/3180c09723c3b61d682732c308c74e2087ac628f/src/libutil/json-utils.hh#L43-L124 you just need this stuff in scope

Comment (Ericson2314) [reply src/drv.cc:78]: https://github.com/nlohmann/json/issues/1749#issuecomment-2498334192 oh huh, the upstream issue finally saw some progress too, and I left this comment.

**Comment by Ericson2314** [comment]
N.B., I think it would be prudent to get Hydra using `nix-eval-jobs` on Nix 2.24, since that version is likely to receive more backports for sake of 24.11. That is no reason not to merge this PR, however. Just it might mean that some future fix needed would be branched off the commit before this lands.

File: src/drv.hh [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #include <nix/get-drvs.hh>
 #include <nix/eval.hh>
 #include <nlohmann/json_fwd.hpp>
+// we need this include or otherwise we cannot instantiate std::optional
+#include <nlohmann/json.hpp> //NOLINT(misc-include-cleaner)
```
Comment (Ericson2314) [inline src/drv.hh:5]: Do we need to do this in `drv.hh` or `drv.cc`? I would think just the latter

**Comment by Mic92** [comment]
I branched out https://github.com/nix-community/nix-eval-jobs/tree/release-2.24 for future nix 2.24 backports for hydra.

File: src/drv.cc (hunk 1 above)
Comment (Mic92) [reply src/drv.cc]: Solved.

**Comment by Mic92** [comment]
@mergify queue

File: src/drv.hh (hunk 2 above)
Comment (Ericson2314) [reply src/drv.hh:5]: Oh, for `Drv::meta`, and implicit constructors

**Pull request merged**